packages/web5-user-agent/src/web5-user-agent.ts

```diff
@@ -190,6 +203,35 @@ export class Web5UserAgent implements Web5Agent {
     };
   }
 
+  async processVcRequest(request: ProcessVcRequest): Promise<VcResponse> {
+    const signedJwt = await this.#sign(request.vc as VerifiableCredential);
+
+    const messageOptions: Partial<RecordsWriteOptions> = { ...{ schema: 'vc/vc', dataFormat: 'application/vc+jwt' } };
+    const { dataBlob, dataFormat } = dataToBlob(signedJwt, messageOptions.dataFormat);
```

Comment:
Similar to the approach in `DwnApi` you'd want to get the data into `Blob` format as early as possible (i.e., do this conversion in `VcApi`) since this is the standard format/encoding we're using when data has to be transported over a network boundary using `fetch()`.  It is unlikely that verifiable credentials would be in the dozens or hundreds of MB in data size, but might as well be consistent.

Here's where it's done in `DwnApi`:
https://github.com/TBD54566975/web5-js/blob/main/packages/web5/src/dwn-api.ts#L327-L342

Reply:
I had to blob in this way because of signing:

I need to sign the vcJWT before I turn it into a blob

Should I create a new method on web5Agent: 
web5Agent.sign()?

or should I send the unsigned blob, un blobify it, sign it, then blobify it again?

Reply:
well actually if I make web5Agent.sign().. I will need to blob it, send it to sign, unblob it, sign it..